iFixit/checkout-ui-extensions
# Refactor: move reactExtension into distinct files

Follow up to #19 from https://github.com/iFixit/checkout-ui-extensions/pull/19#pullrequestreview-1607666201

Shopify seems to encourage `reactExtension(...)` being called in the same file that the component calls.

Summary/CR
--
- Created new files for each extension target that we use, and imported components to use for each of those extensions

I did this as opposed to putting the `reactExtension(...)` in the components because some components show up on both `Thank you` and `Order status` pages. This does mean that when testing components, they will have to be tested separately, since the dev console will let you choose which extension target you want to test.

QA
--
For each target, check that the respective components show up. That is:
- `purchase.checkout.block.render` shows Codice Fiscale and Purchase Order input field (when Purchase Order is selected as payment method)
- `purchase.thank-you.customer-information.render-after` shows what was input into the purchase order field at the `Thank you` page
- `customer-account.order-status.customer-information.render-after` shows the same as above in the `Order status` page
- `customer-account.order-status.block.render` shows the community link on the `Order status` page
- `purchase.thank-you.block.render` shows the same as above on the `Thank you` page

cc: @masonmcelvain
